base/broadcast.jl

```diff
@@ -691,8 +691,48 @@ eltypes(t::Tuple{Any}) = Tuple{_broadcast_getindex_eltype(t[1])}
 eltypes(t::Tuple{Any,Any}) = Tuple{_broadcast_getindex_eltype(t[1]), _broadcast_getindex_eltype(t[2])}
 eltypes(t::Tuple) = Tuple{_broadcast_getindex_eltype(t[1]), eltypes(tail(t)).types...}
 
+function promote_typejoin_union(::Type{T}) where T
+    if T === Union{}
+        return Union{}
+    elseif T isa Union
+        return promote_typejoin(promote_typejoin_union(T.a), promote_typejoin_union(T.b))
+    elseif T <: Tuple
+        return typejoin_union_tuple(T)
+    else
+        return T
+    end
+end
+
+@pure function typejoin_union_tuple(T::Type)
+    p = (Base.unwrap_unionall(T)::Union{DataType, Union}).parameters
```

Comment:
Union doesn't have a `parameters` field—you need to call typejoin_union_tuple on the separate halves

Reply:
```suggestion
    u = Base.unwrap_unionall(T)
    u isa Union && return typejoin(
            typejoin_union_tuple(rewrap_unionall(u.a, T)),
            typejoin_union_tuple(rewrap_unionall(u.b, T)))
    p = (u::DataType).parameters    
```

Reply:
Thanks!